### DISCUSSION TITLE
Unable to receive "auth" parameter of SocketIO client

### DISCUSSION THREAD

**Discussion by mkj-git** [socketio/socket.io#3955] [Q&A] [answered]
This is how i am accessing the SocketIO server

    var socket = io('http://localhost:3001', {
      query: {
        token : "XXXX"
      },
      withCredentials: false,
      allowEIO3: true,
      auth: {
        token: "abc"
      }
    });

IN the backend i am getting the `query` but not the `auth` parameter:

    {"headers":{"host":"localhost:3001","user-agent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0","accept":"*/*","accept-language":"en-US,en;q=0.5","accept-encoding":"gzip, deflate","origin":"http://localhost:4001","dnt":"1","connection":"keep-alive","referer":"http://localhost:4001/","sec-gpc":"1"},"time":"Tue Jun 01 2021 22:29:48 GMT+0530 (India Standard Time)","address":"::ffff:127.0.0.1","xdomain":true,"secure":false,"issued":1622566788895,"url":"/socket.io/?token=XXXX&EIO=3&transport=polling&t=Nd8X5xy","query":{"token":"XXXX","EIO":"3","transport":"polling","t":"Nd8X5xy"}}

**Comment by darrachequesne** [comment, marked as answer]
Hi! I'm not able to reproduce: https://github.com/socketio/socket.io-fiddle/tree/issues/socket.io/3955

Which version of Socket.IO are you using?

**Reply by mkj-git** [reply]
Server side its ` "socket.io": "^2.2.0",` and client side `v2.3.0`

**Reply by darrachequesne** [reply]
Oh, the `auth` option was added in version `3.0.0`.

You can find more information here: https://socket.io/docs/v4/migrating-from-2-x-to-3-0/#Add-a-clear-distinction-between-the-Manager-query-option-and-the-Socket-query-option
